trio/tests/subprocess/test_waitpid_linux.py

```diff
@@ -0,0 +1,48 @@
+import sys
+
+import os
+import pytest
+import signal
+
+from ... import _core
+from ..._subprocess.linux_waitpid import waitpid
+
+pytestmark = pytest.mark.skipif(
+    sys.platform != "linux", reason="linux waitpid only works on linux"
+)
+
+
+async def test_waitpid():
+    pid = os.spawnvp(os.P_NOWAIT, "/bin/false", ("false",))
+    result = await waitpid(pid)
+    # exit code is a 16-bit int: (code, signal)
```

Comment:
In theory it's a bit more complicated than that, there's a "was a core dumped?" flag in there and the ability to distinguish between stop signals and termination signals ([see](https://sourceware.org/git/?p=glibc.git;a=blob;f=bits/waitstatus.h;h=a1c232612cd13e758f222ade6b27bc2c85ab8f8f;hb=HEAD)).

In practice this is a pretty pedantic distinction. If we want to be really POSIX-ly correct, though, I guess the tests should make assertions like

`assert os.WIFEXITED(code) and os.WEXITSTATUS(code) == 1`
`assert os.WIFEXITED(code) and os.WEXITSTATUS(code) == 0`
`assert os.WIFSIGNALED(code) and os.WTERMSIG(code) == signal.SIGKILL`

?